src/widgets/tree/identifier.rs

```diff
@@ -0,0 +1,31 @@
+#![allow(clippy::module_name_repetitions)]
+
+/// Reference to a [`TreeItem`](crate::TreeItem) in a [`Tree`](crate::Tree)
+pub type TreeIdentifier<'a> = &'a [usize];
+/// Reference to a [`TreeItem`](crate::TreeItem) in a [`Tree`](crate::Tree)
+pub type TreeIdentifierVec = Vec<usize>;
```

Comment:
Caveat - I haven't used the tree widget, so I'm looking at it for the first time now.
Perhaps you could model this more explicitly as `struct BranchIdentifier` and `type LeafIdentifier = usize` .

Then `get_without_leaf` method could easily be moved to be a method implemented on `BranchIdentifier`, and rename it to `BranchIdentifier.split_leaf()` or something like that (not sure what would be idiomatic).

My take on performance is that until you measure it, it doesn't really exist. The sorts of projects that render data in a TUI would have approaches to segment a large tree (perhaps) rather than rendering something so huge that cloning identifiers is an issue (this is an assumption - possibly I'm wrong). Either way, I'd suggest pick a node count that represents a realistic upper bound on tree nodes and benchmark that to get a feel for whether you're talking nano seconds or seconds for a clone (or some number in between), and then worrying about it when it's not fast enough for actual use cases that you encounter.